cassovary-core/src/main/scala/com/twitter/cassovary/graph/MemoryMappedDirectedGraph.scala

```diff
@@ -51,48 +70,63 @@ class MemoryMappedDirectedGraph(file: File) extends DirectedGraph[Node] {
     val nodeOutboundOffset = outboundOffset(id)
     val nodeInboundOffset = inboundOffset(id)
-    override def outboundNodes(): Seq[Int] = new IndexedSeq[Int] {
+    def outboundNodes(): Seq[Int] = new IndexedSeq[Int] {
       val length: Int = outDegree(id)
       def apply(i: Int): Int =  data.getInt(nodeOutboundOffset + 4L * i)
-      override def toString(): String = "Memory Mapped IndexedSeq " + mkString("[", ",", "]")
     }
-    override def inboundNodes(): Seq[Int] = new IndexedSeq[Int] {
+    def inboundNodes(): Seq[Int] = new IndexedSeq[Int] {
       val length: Int = inDegree(id)
       def apply(i: Int): Int =  data.getInt(nodeInboundOffset + 4L * i)
-      override def toString(): String = "Memory Mapped IndexedSeq " + mkString("[", ",", "]")
     }
   }
 
-  override def getNodeById(id: Int): Option[Node] =
+  def getNodeById(id: Int): Option[Node] =
     if(0 <= id && id < nodeCount) {
       Some(new MemoryMappedDirectedNode(id))
     } else {
       None
     }
 
-  override def iterator: Iterator[Node] = (0 to nodeCount).iterator flatMap (i => getNodeById(i))
+  def iterator: Iterator[Node] = (0 to nodeCount).iterator flatMap (i => getNodeById(i))
 
-  override def edgeCount: Long = (outboundOffset(nodeCount) - outboundOffset(0)) / 4
+  lazy val edgeCount: Long = outboundOffset(nodeCount) - outboundOffset(0)
 
-  override lazy val maxNodeId: Int = nodeCount - 1
-  // Uncomment in future if maxNodeId becomes a method:
-  // override def maxNodeId = nodeCount - 1
+  override lazy val maxNodeId = nodeCount - 1
 
-  override val storedGraphDir = StoredGraphDir.BothInOut
+  val storedGraphDir = StoredGraphDir.BothInOut
 
-  def loadGraphToRam(): Unit = data.loadFileToRam()
+  /** Loads the graph data into physical RAM.  Makes a "best effort" (see MappedByteBuffer.load()).
+    */
+  def loadToRAM(): Unit = {
+    data.loadFileToRam()
+  }
 }
 
 object MemoryMappedDirectedGraph {
-  /** Writes the given graph to the given file (overwriting it if it exists) in the current binary format.
-    */
+  /** Writes the given graph to the given file (overwriting it if it exists) in the current binary
+   * format.
+   */
   def graphToFile(graph: DirectedGraph[Node], file: File): Unit = {
     val n = graph.maxNodeId + 1 // includes both 0 and maxNodeId as ids
     val out = new DataOutputStream(new BufferedOutputStream(new FileOutputStream(file)))
+    out.writeLong(0)
+    out.writeLong(n)
+    //The outneighbor data starts after the initial 8 bytes, n+1 Longs for outneighbors, and n+1
+    // Longs for in-neighbors
+    var outboundOffset = 16L + 8L * (n + 1) * 2
+    for (i <- 0 until n) {
+      out.writeLong(outboundOffset)
+      outboundOffset += 4 * (graph.getNodeById(i) map (_.outboundCount)).getOrElse(0)
+    }
+    out.writeLong(outboundOffset) // Needed to compute outdegree of node n-1
 
-    def outDegree(id: Int): Int = (graph.getNodeById(id) map (_.outboundCount)).getOrElse(0)
```

Comment:
Good point. I think losing these methods was somehow a side-effect of some other refactoring I did.  I'll restore them.